context/src/env/parser.rs

```diff
@@ -434,3 +533,29 @@ impl<'a> EnvParser<'a> {
         }
     }
 }
+
+#[cfg(feature = "ruby")]
+pub fn ruby_init(ruby: &magnus::Ruby) -> Result<(), magnus::Error> {
+    let ci_platform = ruby.define_class("CIPlatform", ruby.class_object())?;
+    ci_platform.define_method("to_s", magnus::method!(CIPlatform::to_string, 0))?;
+    let ci_info = ruby.define_class("CIInfo", ruby.class_object())?;
+    ci_info.define_singleton_method("new", magnus::function!(CIInfo::new, 1))?;
+    ci_info.define_method("platform", magnus::method!(CIInfo::platform, 0))?;
+    ci_info.define_method("job_url", magnus::method!(CIInfo::job_url, 0))?;
+    ci_info.define_method("branch", magnus::method!(CIInfo::branch, 0))?;
```

Comment:
There is a `define_attr` method of the defined class, but it doesn't work like you'd expect. It makes it so that ruby can read / write to an object constructed in run time, but does not allow you to access the field if it was constructed in rust. Explicit getters and setters seem to be needed.